pkg/controller/utils/utils.go

```diff
@@ -21,6 +18,15 @@ import (
 	routev1 "github.com/openshift/api/route/v1"
 )
 
+// DockerConfigJSON type would be be used to unmarshal pull secrets
+type DockerConfigJSON struct {
```

Comment:
Why is this defined in hive rather than using `k8s.io/kubernetes/pkg/credentialprovider`?

Reply:
We do not vendor `kubernetes` code in Hive. So I think it is fine to just copy the structures.